### PR TITLE
[-] FO: Fix the AuthController.php PHP 5.6 compability

### PR DESCRIPTION
**Pull request by rkorszun**
## Steps to Test this Fix

1. Run presta on php 5.6.19
2. open the login page
   - The page should be opened normally (now it is returning 500)

### CODE REVIEW

**Comment by kpodemski** [comment]
well, you can't access to static using `->`

see #4585

**Comment by rafalkorszun** [comment]
I shouldn't but it will work until PHP 7.0 (that will not trow errors and allow to be backward compatible inside 1.6.x)

File: controllers/front/AuthController.php [hunk 1]
```diff
@@ -361,7 +361,7 @@ protected function processCustomerNewsletter(&$customer)
     {
         $blocknewsletter = Module::isInstalled('blocknewsletter') && $module_newsletter = Module::getInstanceByName('blocknewsletter');
         if ($blocknewsletter && $module_newsletter->active && !Tools::getValue('newsletter')) {
-            if (is_callable(array($module_newsletter, 'isNewsletterRegistered')) && $module_newsletter->isNewsletterRegistered(Tools::getValue('email')) == $module_newsletter::GUEST_REGISTERED) {
+            if (is_callable(array($module_newsletter, 'isNewsletterRegistered')) && $module_newsletter->isNewsletterRegistered(Tools::getValue('email')) == $module_newsletter->GUEST_REGISTERED) {
```
Comment (mickaelandrieu) [inline controllers/front/AuthController.php]: this is not how we access a constant in PHP

**Comment by mickaelandrieu** [comment]
Hi,

@kpodemski is right, closed in favor of #4585.

Thank you for report @rafalkorszun

**Comment by rafalkorszun** [comment]
OK but it isn't integrated on line 1.6.1.x So when you just installing the official (not development) version (via installatron or something) of prestashop  You need  to do some hacks on installation to make working. So for me is till open.

File: controllers/front/AuthController.php (hunk 1 above)
Comment (rafalkorszun) [inline controllers/front/AuthController.php]: yep but this " $module_newsletter::GUEST_REGISTERED" is providing the the "syntax error, unexpected T_PAAMAYIM_NEKUDOTAYIM"  becouse of "$module_newsletter" so your version is providing the 500 error page my is not elegant but will work until 7.0

Comment (mickaelandrieu) [inline controllers/front/AuthController.php]: Can you confirm #4585 fix your issue ?

Comment (rafalkorszun) [inline controllers/front/AuthController.php]: That's Ok if newsletter modules use the same const

**Pull request closed without merging**